muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/fieldio/FileAnswerUtils.java

```
@@ -0,0 +1,151 @@
+  
+  @Inject
+  private WorkspaceMaterialFileFieldAnswerFileDAO workspaceMaterialFileFieldAnswerFileDAO;
+  
+  public Long getLastEntityId() {
+    SystemSetting idSetting = systemSettingDAO.findByKey(SETTING_LASTID);
+    return idSetting == null ? 0L : Long.valueOf(idSetting.getValue());
+  }
+  
+  public void setLastEntityId(Long id) {
+    SystemSetting idSetting = systemSettingDAO.findByKey(SETTING_LASTID);
+    if (idSetting == null) {
+      systemSettingDAO.create(SETTING_LASTID, id.toString());
+    }
+    else {
+      systemSettingDAO.update(idSetting, id.toString());
+    }
+  }
+  
+  public boolean isFileSystemStorageEnabled() {
+    SystemSetting enabledSetting = systemSettingDAO.findByKey(SETTING_ENABLED);
+    return enabledSetting != null && Boolean.valueOf(enabledSetting.getValue());
+  }
+  
+  public byte[] getFileContent(Long userEntityId, String fileId) throws IOException {
+    if (userEntityId == null || fileId == null) {
+      logger.severe(String.format("Invalid fetch parameters: file %s of user %d", fileId, userEntityId));
+      return new byte[0];
+    }
+    SystemSetting fileFolderSetting = systemSettingDAO.findByKey(SETTING_FILEFOLDER);
+    if (fileFolderSetting != null && !StringUtils.isBlank(fileFolderSetting.getValue())) {
+      java.io.File answerFile = Paths.get(fileFolderSetting.getValue(), userEntityId.toString(), fileId).toFile();
+      if (answerFile.exists()) {
+        return Files.readAllBytes(answerFile.toPath());
+      }
+    }
+    logger.warning(String.format("Failed to retrieve file %s of user %d", fileId, userEntityId));
+    return new byte[0];
+  }
+  
+  public void storeFileToFileSystem(Long userEntityId, String fileId, byte[] content) throws IOException {
+    if (userEntityId == null || fileId == null || content == null) {
+      logger.severe(String.format("Invalid store parameters or no file content: file %s of user %d", fileId, userEntityId));
+      return;
```

Probably needs to be exception here to indicate the file couldn't be saved at all.

---

Changed calling side to catch all exceptions